api/server/sdk/cloud_backup.go

```diff
@@ -0,0 +1,252 @@
+/*
+Package sdk is the gRPC implementation of the SDK gRPC server
+Copyright 2018 Portworx
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package sdk
+
+import (
+	"context"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/libopenstorage/openstorage/api"
+	"github.com/libopenstorage/openstorage/volume"
+)
+
+// CloudBackupServer is an implementation of the gRPC OpenStorageCloudBackup interface
+type CloudBackupServer struct {
+	driver volume.VolumeDriver
+}
+
+// Create creates a backup for a volume
+func (s *CloudBackupServer) Create(
+	ctx context.Context,
+	req *api.SdkCloudBackupCreateRequest,
+) (*api.SdkCloudBackupCreateResponse, error) {
+
+	if len(req.GetVolumeId()) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Must supply a volume id")
+	} else if len(req.GetCredentialUuid()) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Must supply credential uuid")
+	}
+
+	// Create the backup
+	if err := s.driver.CloudBackupCreate(&api.CloudBackupCreateRequest{
+		VolumeID:       req.GetVolumeId(),
+		CredentialUUID: req.GetCredentialUuid(),
+		Full:           req.GetFull(),
+	}); err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to create backup: %v", err)
+	}
+
+	return &api.SdkCloudBackupCreateResponse{
+		// THIS WILL BE FIXED BY #### where Create will return an ID
+		BackupId: "TBD",
+	}, nil
+}
+
+// Restore a backup
+func (s *CloudBackupServer) Restore(
+	ctx context.Context,
+	req *api.SdkCloudBackupRestoreRequest,
+) (*api.SdkCloudBackupRestoreResponse, error) {
+
+	if len(req.GetBackupId()) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Must provide backup id")
+	} else if len(req.GetCredentialUuid()) == 0 {
```

Comment:
nit: do we really need else statements 
couple of if's would do , idiomatic go.

Reply:
Yeah, I understand what you mean, but it is ok.